examples/gno.land/r/demo/boards2/public.gno

```diff
@@ -57,21 +72,38 @@ func CreateReply(bid BoardID, threadID, replyID PostID, body string) PostID {
 	board := mustGetBoard(bid)
 	thread := mustGetThread(board, threadID)
 
+	assertThreadVisible(thread)
+
 	// TODO: Assert thread is not locked
 	// TODO: Assert that caller is a board member (when board type is invite only)
-
 	var reply *Post
 	if replyID == threadID {
 		// When the parent reply is the thread just add reply to thread
 		reply = thread.AddReply(caller, body)
 	} else {
 		// Try to get parent reply and add a new child reply
 		post := mustGetReply(thread, replyID)
+		assertThreadVisible(thread)
+
 		reply = post.AddReply(caller, body)
 	}
 	return reply.id
 }
 
+func FlagReply(bid BoardID, threadID, replyID PostID, reason string) {
+	caller := std.GetOrigCaller()
+
+	board := mustGetBoard(bid)
+	assertHasBoardPermission(board, caller, PermissionThreadFlag)
+
+	thread := mustGetThread(board, threadID)
+	reply := mustGetReply(thread, replyID)
+
+	if flagItem(reply, NewFlag(caller, reason)) {
+		reply.SetVisible(false)
+	}
```

Comment:
Maybe adding a `hide` variable makes it more semantic by making it explicit that flagging an item might in some cases hide the reply. Without the variable one could think that the reply would be hidden after it is flagged successfully.

```suggestion
	if hide := flagItem(reply, NewFlag(caller, reason)); hide {
		reply.SetVisible(false)
	}
```